lineapy/plugins/pipeline_writers.py

```diff
@@ -173,15 +174,92 @@ def _write_dag(self) -> None:
                 MAX_ACTIVE_RUNS=self.dag_config.get("max_active_runs", 1),
                 CATCHUP=self.dag_config.get("catchup", "False"),
                 tasks=session_functions,
-                task_dependencies=task_graph.get_airflow_dependency(),
+                task_dependencies=task_graph.get_airflow_dependencies(),
+            )
+            full_code = prettify(full_code)
+
+            # Write out file
+            file = self.output_dir / f"{self.pipeline_name}_dag.py"
+            file.write_text(prettify(full_code))
+            logger.info("Generated DAG file %s", file)
```

Comment:
as much as possible, lets avoid repeating code. this piece can be pulled outside the if. the else block raises exception so it will not reach the write part.